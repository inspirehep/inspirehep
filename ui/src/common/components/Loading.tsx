import React, { Component } from 'react';
import { Row, Col, Spin } from 'antd';

class Loading extends Component {
  render() {
    return (
      <Row className="w-100" type="flex" justify="center" align="middle">
        <Col>
          <div data-test-id="loading" className="tc pa4">
            <Spin tip="Loading ..." />
          </div>
        </Col>
      </Row>
    );
  }
}

export default Loading;
