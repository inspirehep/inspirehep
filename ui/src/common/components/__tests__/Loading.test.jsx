import React from 'react';
import { shallow } from 'enzyme';

import Loading from '../Loading';

describe('Loading', () => {
  it('render loading component', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
