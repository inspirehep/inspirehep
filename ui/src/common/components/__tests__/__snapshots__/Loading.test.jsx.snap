// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading render loading component 1`] = `
<Row
  align="middle"
  className="w-100"
  gutter={0}
  justify="center"
  type="flex"
>
  <Col>
    <div
      className="tc pa4"
      data-test-id="loading"
    >
      <Spin
        size="default"
        spinning={true}
        tip="Loading ..."
        wrapperClassName=""
      />
    </div>
  </Col>
</Row>
`;
